### PR TITLE
[shape_infer]Implement shape inference of Roll, ROIAlign,Proposal

### PR DESCRIPTION
**Pull request by luweizhou2016**
### Details:
 - *ExperimentalDetectronGenerateProposalsSingleImage*
 - *Proposal*
 - *Roll*
 - *ROIAlign*

### Tickets:
 - *66209*
 - *69048*
 - *69049*
 - *69052*

### CODE REVIEW

**Comment by luweizhou2016** [comment]
@v-Golubev, can you pls also help to review this PR? Thx!

**Review by v-Golubev — COMMENTED** [review]
Please take a look on comments

File: src/core/shape_inference/include/proposal_shape_inference.hpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+template <class T>
+void shape_infer(const ov::op::v4::Proposal* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 3 && output_shapes.size() == 2);
+
+    const auto proposal_v0_op = dynamic_cast<const ov::op::v0::Proposal*>(op);
+    auto output_vector_v0 = std::vector<T>{output_shapes[0]};
+    shape_infer(proposal_v0_op, input_shapes, output_vector_v0);
+    output_shapes[0] = output_vector_v0[0];
+
```
[the remark this answers is not in the document]
Comment (luweizhou2016) [reply src/core/shape_inference/include/proposal_shape_inference.hpp:99]: Thx. I like the the idea it makes code pretty clean. I have made minor change for the interface 3rd parameter to 
std::vector<ShapeType>& output_shapes

File: src/core/shape_inference/include/proposal_shape_inference.hpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    const auto& proposals_ps = output_vector_v0[0];
+    auto& out_ps = output_shapes[1];
+    if (proposals_ps.rank().is_static() && proposals_ps.size() == 2) {
+        out_ps = T{proposals_ps[0]};
+    } else {
```
[the remark this answers is not in the document]
Comment (luweizhou2016) [reply src/core/shape_inference/include/proposal_shape_inference.hpp:104]: Yes. Also Directly changed to out_ps = T{proposals_ps[0]}; No needs to check.

**Comment by luweizhou2016** [comment]
@v-Golubev , Thanks for the detailed review. Review comments applied. Can you pls help to review when available?

**Comment by luweizhou2016** [comment]
@v-Golubev , Applied review comments. Thx!

**Review by v-Golubev — APPROVED** [review]
Changes look good to me, thanks!

File: src/core/shape_inference/include/roll_shape_inference.hpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                                      ", axis: ",
+                                      axis);
+                if (axis < 0) {
+                    axis += static_cast<int64_t>(data_rank);
+                }
+                NODE_VALIDATION_CHECK(op,
+                                      axis >= 0,
+                                      "Axes must be positive or equal to zero. Got "
+                                      "axis: ",
+                                      axis);
```
Comment (jane-intel) [inline src/core/shape_inference/include/roll_shape_inference.hpp:63]: nit: there is a helper for that already in validation utils

**Review by jane-intel — APPROVED** [review]

**Pull request merged**